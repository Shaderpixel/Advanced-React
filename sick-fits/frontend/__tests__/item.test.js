import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';
import AddtoCartComponent from '../components/AddToCart';
import DeleteItemComponent from '../components/DeleteItem';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 50000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largeDog.jpg',
};

describe('<Item />', () => {
  // it('render the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   // console.log(img.props());
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });
  // it('render the title and PriceTag properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   // console.log(PriceTag.children().text());
  //   // console.log(PriceTag.dive().text());
  //   // console.log(wrapper.debug());
  //   // console.log(PriceTag.debug());
  //   expect(PriceTag.children().text()).toBe('$500');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });
  // it('render out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   // console.log(wrapper.debug());
  //   const buttonList = wrapper.find('.buttonList');
  //   console.log(buttonList.children().debug());
  //   expect(buttonList.children()).toHaveLength(3);
  //   // check to see if the button components exist
  //   expect(
  //     buttonList.children().containsMatchingElement(DeleteItemComponent)
  //   ).toEqual(true);
  //   // or check if the button selectors are there
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('Link').exists()).toBe(true);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  //   expect(buttonList.find('Link')).toBeTruthy();
  // });
  // snapshot testing
  it('render and match the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
