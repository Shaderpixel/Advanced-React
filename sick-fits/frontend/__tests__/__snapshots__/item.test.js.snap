// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> render and match the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $500
  </PriceTag>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
      itemDetails={
        Object {
          "description": "This item is really cool!",
          "id": "ABC123",
          "image": "dog.jpg",
          "largeImage": "largeDog.jpg",
          "price": 50000,
          "title": "A Cool Item",
        }
      }
    />
    <DeleteItem
      id="ABC123"
    >
      Delete This Item 🗑
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
